Builds/CMake/CMakeFuncs.cmake

```diff
@@ -502,6 +498,15 @@ macro(setup_build_boilerplate)
       APPEND
       PROPERTY COMPILE_DEFINITIONS
       $<$<OR:$<BOOL:${profile}>,$<CONFIG:Release>,$<CONFIG:ReleaseClassic>>:NDEBUG>)
+  else()
+    # CMAKE_CXX_FLAGS_RELEASE is created by CMake for most / all generators
+    #  with defaults including /DNDEBUG or -DNDEBUG, and that value is stored
+    #  in the cache. Override that locally so that the cache value will be
+    #  avaiable if "assert" is ever changed.
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASECLASSIC "${CMAKE_CXX_FLAGS_RELEASECLASSIC}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASECLASSIC "${CMAKE_C_FLAGS_RELEASECLASSIC}")
```

Comment:
I originally had `[-/]DNDEBUG`, but I made it more permissive in case one of the other generators uses a different command line flag. I don't know of any, but there are a lot of generators. If `[-/]D` is safe to assume as a universal standard, I'll change it back.

`LIST(REMOVE_ITEM` doesn't work, because these are not CMake lists (`;`-delimited). They're more like command line params (space-delimited). Eg.
```
$ grep CXX_FLAGS_RELEASE: build/cmake/CMakeCache.txt
CMAKE_CXX_FLAGS_RELEASE:STRING=/MD /O2 /Ob2 /DNDEBUG
```